### PR TITLE
Add /about/indexing endpoint

### PR DESCRIPTION
**Pull request by Uxio0**
- Refactor ERC20Indexing endpoint into service

### CODE REVIEW

File: safe_transaction_service/history/services/index_service.py [hunk 1]
```diff
@@ -24,6 +25,23 @@
 logger = logging.getLogger(__name__)
 
 
+@dataclass
+class IndexingStatus:
+    current_block_number: int
+    erc20_block_number: int
+    erc20_synced: bool
+    master_copies_block_number: bool
```
Comment (moisses89) [inline safe_transaction_service/history/services/index_service.py:33]: int ;)

File: safe_transaction_service/history/services/index_service.py [hunk 2]
```diff
@@ -88,6 +106,51 @@ def block_get_or_create_from_block_hash(self, block_hash: int):
                 block, confirmed=confirmed
             )
 
+    def get_indexing_status(self) -> IndexingStatus:
+        current_block_number = self.ethereum_client.current_block_number
+        erc20_block_number = SafeContract.objects.aggregate(
+            min_erc20_block_number=Min("erc20_block_number")
+        )["min_erc20_block_number"]
+        if erc20_block_number is None:  # Still nothing indexed
+            erc20_block_number = current_block_number
```
Comment (moisses89) [inline safe_transaction_service/history/services/index_service.py:115]: I didn't get this.
`current_block_number` is last block indexed? and `erc20_block_number` is last indexed block, I'd said that the last block indexed should be max instead min, but I'm sure that I'm missing something.

Comment (Uxio0) [reply safe_transaction_service/history/services/index_service.py:115]: `current_block_number` is the current block on the chain, and `erc20_block_number` is the block where erc20 indexing is stuck, so, the minimum block number for all the Safes

File: safe_transaction_service/history/services/index_service.py (hunk 1 above)
Comment (Uxio0) [reply safe_transaction_service/history/services/index_service.py]: Fixed

File: safe_transaction_service/history/services/index_service.py (hunk 2 above)
Comment (moisses89) [reply safe_transaction_service/history/services/index_service.py:115]: Ok, then `current_block_number` means the position where we are on chain, and for this reason
`if erc20_block_number is None -> erc20_block_number = current block.` that means that there wasn't anything to index until such `current_block_number`, true?

File: safe_transaction_service/history/tests/test_views.py [hunk 3]
```diff
@@ -93,29 +94,119 @@ def test_erc20_indexing_view(self, current_block_number_mock: PropertyMock):
...
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["current_block_number"], 2000)
+        self.assertEqual(response.data["erc20_block_number"], 10)
+        self.assertEqual(response.data["erc20_synced"], False)
+
+        SafeContractFactory(erc20_block_number=12)
+        response = self.client.get(url, format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["current_block_number"], 2000)
+        self.assertEqual(response.data["erc20_block_number"], 10)
```
Comment (moisses89) [inline safe_transaction_service/history/tests/test_views.py:119]: Shouldn't return 12?

Comment (Uxio0) [reply safe_transaction_service/history/tests/test_views.py:119]: No, as the minimum is 10

**Review by moisses89 — APPROVED** [review]

**Pull request merged**